### PR TITLE
Improve test coverage for pydantic.functional_validator

### PR DESCRIPTION
**Pull request by hramezani**
Selected Reviewer: @dmontagu

### CODE REVIEW

File: tests/test_validators.py [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
 
 def test_annotated_validator_after() -> None:
-    MyInt = Annotated[int, AfterValidator(lambda x: x if x != -1 else 0)]
+    MyInt = Annotated[int, AfterValidator(lambda x, _info: x if x != -1 else 0)]
```
Comment (hramezani) [inline tests/test_validators.py:35]: This change here covers [this branch in AfterValidator.__get_pydantic_core_schema__](https://github.com/pydantic/pydantic/blob/cf48b1a5a494604ec161ed85df6285d683084baa/pydantic/functional_validators.py#L32).

File: tests/test_validators.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ class Model(BaseModel):
 
 
 def test_annotated_validator_before() -> None:
-    FloatMaybeInf = Annotated[float, BeforeValidator(lambda x: x if x != 'zero' else 0.0)]
+    FloatMaybeInf = Annotated[float, BeforeValidator(lambda x, _info: x if x != 'zero' else 0.0)]
```
Comment (hramezani) [inline tests/test_validators.py:48]: same as `AfterValidator` but cover the same branch in `BeforeValidator`

File: tests/test_validators.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ class Model(BaseModel):
 
 
 def test_annotated_validator_plain() -> None:
-    MyInt = Annotated[int, PlainValidator(lambda x: x if x != -1 else 0)]
+    MyInt = Annotated[int, PlainValidator(lambda x, _info: x if x != -1 else 0)]
```
Comment (hramezani) [inline tests/test_validators.py:59]: same as `AfterValidator` but cover the same branch in `PlainValidator `

**Review by lig — APPROVED** [review]
👍🏻

**Review by dmontagu — APPROVED** [review]

**Pull request merged**